### PR TITLE
SOLR-16397: Modify v2 'MERGEINDEXES' API to be more REST-ful

### PR DESCRIPTION
**Pull request by iamsanjay**
https://issues.apache.org/jira/browse/SOLR-16397

# Description

V2(JAX-RS) Endpoints for core admin op "MERGEINDEXES".

# Solution

This commit changes the v2 "MERGEINDEXES" API to be more in line with the REST-ful design we're targeting for Solr's v2 APIs.

Following these changes, the v2 API now appears as: `POST /api/cores/coreName/merge-indices {...}`

# Tests

Three test cases added:
1. Both indexDir and srcCore are empty. 
2. srcCore does not exists.
3. indexDir path not allowed.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [x] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/handler/admin/api/MergeIndexes.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  public MergeIndexes(
+      CoreContainer coreContainer,
+      CoreAdminHandler.CoreAdminAsyncTracker coreAdminAsyncTracker,
+      SolrQueryRequest req,
+      SolrQueryResponse rsp) {
+    super(coreContainer, coreAdminAsyncTracker, req, rsp);
+  }
+
+  @Override
+  public SolrJerseyResponse mergeIndexes(String coreName, MergeIndexesRequestBody requestBody)
```
Comment (iamsanjay) [inline solr/core/src/java/org/apache/solr/handler/admin/api/MergeIndexes.java:66]: Permission is missing in the old V2 API class, however, I believe this command is changing the index for the given core name and that's why we should add the permission (EDIT) annotation here as well.

**Comment by gerlowskija** [comment]
Hey - sorry for the delay in reviewing this over the holidays.  Taking a look now!

**Review by gerlowskija — COMMENTED** [review]
A handful of minor comments, but otherwise this looks great.  Lmk what you're planning to address when you get a chance to read through and we can go from there

File: solr/api/src/java/org/apache/solr/client/api/model/MergeIndexesRequestBody.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.v3.oas.annotations.media.Schema;
+import java.util.List;
+
+/** Request body for endpoints {@link org.apache.solr.client.api.endpoint.MergeIndexesApi} */
+public class MergeIndexesRequestBody {
+  @Schema(description = "Multi-valued, directories that would be merged.")
+  @JsonProperty
+  public List<String> indexDir;
```
[the remark this answers is not in the document]
Comment (iamsanjay) [reply solr/api/src/java/org/apache/solr/client/api/model/MergeIndexesRequestBody.java:27]: Yup, make more sense! But I only changed the java variable names, keeping the older request parameter names . Or, do you want me to change those as well?

[the remark this answers is not in the document]
Comment (gerlowskija) [reply solr/api/src/java/org/apache/solr/client/api/model/MergeIndexesRequestBody.java]: I think it makes sense to rename both.  Backcompat doesn't apply to the v2 API yet, so it's best to improve this now while we have the chance.

**Comment by gerlowskija** [comment]
> I saw some APIs override the below method to mark it expensive. I believe, MERGEINDEXES is kind of expensive.

+1 I think it makes sense to give `MergeIndexes` its own `isExpensive()` method returning `true`, good catch!

**Pull request merged**